db/system_keyspace.cc

```diff
@@ -2649,6 +2651,23 @@ future<std::vector<system_keyspace::view_build_progress>> system_keyspace::load_
     });
 }
 
+static future<std::optional<mutation>> get_scylla_local_mutation(replica::database& db, std::string_view key) {
+    auto s = db.find_schema(db::system_keyspace::NAME, db::system_keyspace::SCYLLA_LOCAL);
+
+    partition_key pk = partition_key::from_singular(*s, key);
+    dht::partition_range pr = dht::partition_range::make_singular(dht::decorate_key(*s, pk));
+
+    auto rs = co_await replica::query_mutations(db.container(), s, pr, s->full_slice(), db::no_timeout);
+    SCYLLA_ASSERT(rs);
```

Comment:
Too strict.

Reply:
I've only moved this function up, SCYLLA_ASSERT() was already there.
But is it that bad? As I see, `replica::query_mutations()` should always return valid pointer, so isn't assert correct in this case?

Reply:
extracted to separate commit